ai21_tokenizer/jurassic_tokenizer.py

```diff
@@ -20,250 +20,163 @@ class SpaceSymbol:
     count: int
 
 
-class JurassicTokenizer(BaseTokenizer):
+class JurassicTokenizer(BaseJurassicTokenizer, BaseTokenizer):
     def __init__(
         self,
         model_path: Optional[PathLike] = None,
         config: Optional[Dict[str, Any]] = None,
         model_file_handle: Optional[BinaryIO] = None,
     ):
-        self._validate_init(model_path=model_path, model_file_handle=model_file_handle)
-
+        BaseJurassicTokenizer.__init__(self, model_path=model_path, config=config, model_file_handle=model_file_handle)
         model_proto = load_binary(self._get_model_file(model_path)) if model_path else model_file_handle.read()
-
         # noinspection PyArgumentList
         self._sp = spm.SentencePieceProcessor(model_proto=model_proto)
-        config = self._get_config(model_path=model_path, config=config)
-
-        self.pad_id = config.get("pad_id")
-        self.unk_id = config.get("unk_id")
-        self.eop_id = config.get("eop_id")
-        self.bos_id = config.get("bos_id")
-        self.eos_id = config.get("eos_id")
-
-        self._newline_piece = config.get("newline_piece")
-        self._mask_pieces = config.get("mask_pieces", [])
-
-        self._manual_add_dummy_prefix = not (config.get("add_dummy_prefix", True))
-
         self._id_to_token_map = {i: self._sp.id_to_piece(i) for i in range(self.vocab_size)}
         self._token_to_id_map = {self._sp.id_to_piece(i): i for i in range(self.vocab_size)}
         self._no_show_tokens = set(
             self._convert_ids_to_tokens([i for i in range(self.vocab_size) if self._sp.IsControl(i)])
         )
```

Comment:
All of these lines are already happening in the base class no?

Reply:
Nope.
In base class are all the initializations that don't depend on load_binary, so that both sync and async classes can utilize.
These lines are for the initialization of the members that depend on load_binary (the "heavy" operation, that happens in async in the async class).